src/Credentials/InstanceProfileProvider.php

```diff
@@ -24,18 +25,27 @@ class InstanceProfileProvider
     /** @var callable */
     private $client;
 
+    /** @var int */
+    private $retries;
+
+    /** @var int */
+    private $attempts;
+
     /**
      * The constructor accepts the following options:
      *
      * - timeout: Connection timeout, in seconds.
      * - profile: Optional EC2 profile name, if known.
+     * - retries: Optional number of retries to be attempted.
      *
      * @param array $config Configuration options.
      */
     public function __construct(array $config = [])
     {
         $this->timeout = isset($config['timeout']) ? $config['timeout'] : 1.0;
         $this->profile = isset($config['profile']) ? $config['profile'] : null;
+        $this->retries = isset($config['retries']) ? $config['retries'] : 0;
```

Comment:
Should this retry by default?

Reply:
Once?  Or preserve the existing behavior of this not getting retried by default.

Reply:
3 sounds like a good compromise.